daemon/algod/api/server/v2/test/handlers_test.go

```diff
@@ -551,6 +552,45 @@ func TestTealCompile(t *testing.T) {
 	tealCompileTest(t, badProgramBytes, 400, true)
 }
 
+func tealDisassembleTest(t *testing.T, stringToUse string, expectedCode int, enableDeveloperAPI bool) {
+	numAccounts := 1
+	numTransactions := 1
+	offlineAccounts := true
+	mockLedger, _, _, _, releasefunc := testingenv(t, numAccounts, numTransactions, offlineAccounts)
+	defer releasefunc()
+	dummyShutdownChan := make(chan struct{})
+	mockNode := makeMockNode(mockLedger, t.Name(), nil)
+	mockNode.config.EnableDeveloperAPI = enableDeveloperAPI
+	handler := v2.Handlers{
+		Node:     mockNode,
+		Log:      logging.Base(),
+		Shutdown: dummyShutdownChan,
+	}
+	e := echo.New()
+	req := httptest.NewRequest(http.MethodPost, "/", bytes.NewReader([]byte(stringToUse)))
+	rec := httptest.NewRecorder()
+	c := e.NewContext(req, rec)
+	err := handler.TealDisassemble(c)
+	require.NoError(t, err)
+	require.Equal(t, expectedCode, rec.Code)
+}
+
+func TestTealDisassemble(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	// nil program works, but results in invalid version text.
+	tealDisassembleTest(t, "", 200, true)
+	goodProgram := `int 1`
```

Comment:
I recommend adding more good and bad examples.

For good examples, I also recommend:
* asserting round trips (i.e. program with no optimizations --assembler--> assembled base64 --disassemble endpoint--> original program)
* _OR_ asserting that the disassembled response is the same as what the disassembler gives

For bad examples, I also recommend asserting the error message gleaned from the response is what you want it to be, in addition to producing the correct http status.

Reply:
Good point, I added checks to assert that the error messages and the response messages are what we expect. I was looking into adding more error messages, but they are actually quite tricky to test with respect to the disassembler and require manipulating some raw bytes, which looks a bit messy to me, so I left it as is. Let me know if you have any further suggestions.

Reply:
The most recent test-code changes are a good addition.

I recommend even more`goodProgram`'s and `badPrograms`'s. In particular, it would be nice to try programs from all avaialable versions, and possibly an extra emphasis on the current logic version.

Reply:
I agree that this causes some test duplication. My philosophy is that it doesn't harm to test for various scenarios from the point of view of a consumer, even if this causes duplication. But I don't feel strongly about this.

Reply:
I updated the tests a bit to reflect the API changes (takes in a binary instead of a base64 encoded program). I'm pretty happy with the test coverage right now, but let me know if you have any other thoughts!